### PR TITLE
Feat: Automation design-to-code

### PR DESCRIPTION
**Pull request by afreire-laptop**
## Description

This PR adds Automation from Design-to-code:
 - Automated generation of Themes from JSON token files
 - GitHub Workflow that runs specific for Themes generation
 - Dedicated package for Themes (outside of React scope)

Theme Generation workflow:
 - Download of JSON Token files from Figma Variables. We use of the following Figma [Plugin](https://www.figma.com/community/plugin/1263743870981744253/design-tokens-manager)  which is compliant with W3C Design Token draft)
 - Open new branch
 - Add JSON Tokens to dedicated `themes/tokens` folder
 - Commit and push changes
 

## Issue

[DTUI-76](https://jir.t3.daimlertruck.com/browse/DTUI-76)

## Type of change

- [x] Feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Refactor
- [ ] Chore
- [ ] Documentation
- [ ] Tests
- [ ] Other (please specify):

## Check requirements

- [x] The code follows the project's coding standards and guidelines
- [x] Documentation is updated accordingly
- [x] All existing tests are passing
- [x] I have added new tests to cover the changes

### CODE REVIEW

File: .github/workflows/deploy-storybook.yml [hunk 1]
```diff
@@ -35,6 +35,9 @@ jobs:
         with:
           CACHE_FILE_NAME: 'global'
 
+      - name: Build dependencies
+        run: yarn run turbo build --filter='./packages/*'
+
```
Comment (catiaraminhos) [inline .github/workflows/deploy-storybook.yml:40]: Just curiosity: was the storybook deployment working before this change?

File: packages/themes/README.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+Includes a GitHub Actions workflow (`.github/workflows/theme-generation.yml`) that automatically generates themes when token files are modified in pull requests.
+
+#### How it works:
+
+1. **Trigger**: The workflow runs when a PR is opened, updated, or reopened and contains changes to:
+   - `packages/react-packages/core/src/tokens/**/*.json`
+   - `packages/react-packages/core/src/utils/theme-generator/**/*`
+   - `packages/react-packages/core/src/types/theme.ts`
+   - `packages/react-packages/core/package.json`
```
Comment (catiaraminhos) [inline packages/themes/README.md:64]: Shouldn't be `packages/themes/...`?

File: .github/workflows/deploy-storybook.yml (hunk 1 above)
Comment (afreire-laptop) [reply .github/workflows/deploy-storybook.yml:40]: Yes. That's because of the dependency chain from @dt-dds/themes and the workflow doesn't know it needs to build packages/themes because the path isn't in the trigger list - https://github.com/daimlertruck/DT-DDS/blob/feat/DTUI-76/automation-design-to-code/.github/workflows/deploy-storybook.yml#L16-L20

I still don't want to add the themes path to the trigger list because it will cause a race condition with theme-generation.yml workflow, but eventually we'll solve that also. Anyway, having the `Build deps` step is the wright way to avoid breaking the CI

File: packages/themes/src/tokens/color.styles.tokens.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+          "position": 0.55
+        },
+        {
+          "color": "#dedede00",
+          "position": 1
+        }
+      ]
+    }
+  }
+}
```
Comment (catiaraminhos) [inline packages/themes/src/tokens/color.styles.tokens.json:23]: nitpicking: All tokens JSON files are missing an empty line at the end of the file. Should we add the empty line?

File: packages/themes/src/themes/daimler-truck/colors.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import { CustomTheme as Theme } from '../../types';
+
+export const colors: Theme['colors'] = {
+  grey_00: '#ffffff',
+  grey_10: '#fafafa',
```
[the remark this answers is not in the document]
Comment (afreire-laptop) [reply packages/themes/src/themes/daimler-truck/colors.ts:5]: Yes. Figma tokens have been updated to reflect DT guidelines. That still needs to be reviewed because it may impact some components. Not a refactor but a new PR replacing the tokens files.

File: packages/themes/src/tokens/Themes.DT-Companion.tokens.json [hunk 5]
```diff
@@ -0,0 +1,3661 @@
+{
```
[the remark this answers is not in the document]
Comment (afreire-laptop) [reply packages/themes/src/tokens/Themes.DT-Companion.tokens.json]: It's a good point... Those themes still exist in our Figma tokens (that was why). I'll dropped them for now. 
But bare in mind that theme generation still needs to run from this codebase. So if our products still use those custom themes, they'll still need to generate them here. 

In the future, we can think about exposing this theme-generator functionality to be used directly in the product codebases.

File: packages/react-packages/toast/src/Toast.test.tsx [hunk 6]
```diff
@@ -198,4 +217,441 @@ describe('emitToast', () => {
...
+  });
+
+  it('should test the conditional child rendering in Children.map', () => {
+    const ProvidedToaster = withProviders(Toaster);
+
+    const mockProps = {
+      type: ToastType.Success,
+      title: 'Conditional Child Test',
+      message: 'Testing conditional child rendering',
+      children: false ? (
```
[the remark this answers is not in the document]
Comment (afreire-laptop) [reply packages/react-packages/toast/src/Toast.test.tsx:607]: It's evaluating it renders children correctly based on conditional expressions. It covers an edge case. Might be a bit pointless but it doesn't hurt, right?

File: packages/react-packages/toast/src/Toast.test.tsx [hunk 7]
```diff
@@ -198,4 +217,441 @@ describe('emitToast', () => {
...
+
+    act(() => {
+      emitToast(mockProps);
+    });
+
+    expect(screen.getByText('Nested Button')).toBeInTheDocument();
+    expect(screen.getByText('Nested Text')).toBeInTheDocument();
+  });
+
+  it('should test the Children.map null check logic', () => {
```
[the remark this answers is not in the document]
Comment (afreire-laptop) [reply packages/react-packages/toast/src/Toast.test.tsx]: Dropped!

File: packages/themes/src/tokens/Themes.DT-Companion.tokens.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (catiaraminhos) [reply packages/themes/src/tokens/Themes.DT-Companion.tokens.json]: True, while we don't have a way to generate the themes outside of this codebase, they still need to be generated here.

File: packages/themes/README.md [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+
+## Theme Generation
+
+The theme generation system automatically processes JSON token files and generates corresponding TypeScript theme files that match the `CustomTheme` type structure. It supports multiple themes and handles token validation, reference resolution, and fallback values.
+
+### JSON token files
+
+The JSON token files are being generated from Figma Variables, with the use of the following plugin [Design Token Manager](https://www.figma.com/community/plugin/1263743870981744253/design-tokens-manager).
+
+The exported JSON file format follows the spec of the [Design Tokens format](https://www.designtokens.org/tr/drafts/format/) purposed by the W3C work group.
```
[the remark this answers is not in the document]
Comment (catiaraminhos) [reply packages/themes/README.md:50]: Thanks for the context!

**Review by catiaraminhos — APPROVED** [review]
